Source/SharpDX.XAudio2/VoiceShadow.cs

```diff
@@ -64,50 +64,80 @@ public VoiceVtbl()
             static private void OnVoiceProcessingPassStartImpl(IntPtr thisObject, int bytes)
             {
                 var shadow = ToShadow<VoiceShadow>(thisObject);
+                if (shadow == null)
```

Comment:
Coding style only comments.  

Would you not use "If (Shadow != null)" and nest the logic?  

Or would you also consider using and I have not tested this, so it might not be valid -

var callback = (VoiceCallback)shadow**?**.Callback;
callback**?**.OnVoiceProcessingPassStart(bytes);

only a passing question.